### PR TITLE
Add a new exception type specifically for jira component name not found exceptions

### PR DESCRIPTION
**Pull request by halkeye**

### CODE REVIEW

**Comment by halkeye** [comment]
image

yay! seems to split them up!

File: src/main/java/io/jenkins/plugins/models/MissingJiraComponentException.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    return componentName;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (!(o instanceof MissingJiraComponentException)) return false;
+    MissingJiraComponentException that = (MissingJiraComponentException) o;
+    return pluginName.equals(that.pluginName) &&
+      componentName.equals(that.componentName);
```
Comment (timja) [inline src/main/java/io/jenkins/plugins/models/MissingJiraComponentException.java:29]:
```suggestion
   return Objects.equals(pluginName, that.pluginName) && Objects.equals(componentName, that.componentName);
```

Comment (halkeye) [reply src/main/java/io/jenkins/plugins/models/MissingJiraComponentException.java:29]: Fair enough. I don't think that's possible but it's a good safety-net. It's auto generated code anyways

Comment (halkeye) [reply src/main/java/io/jenkins/plugins/models/MissingJiraComponentException.java:29]: Fair enough. I don't think that's possible but it's a good safety-net. It's auto generated code anyways

**Comment by halkeye** [comment]
KK. I'll sick the ide on it in the morning

**Comment by timja** [comment]
> KK. I'll sick the ide on it in the morning

Created https://github.com/jenkins-infra/plugin-site-api/issues/90 😄

**Pull request merged**